sql/expression/function/lower_upper.go

```diff
@@ -0,0 +1,120 @@
+package function
+
+import (
+	"fmt"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+	"gopkg.in/src-d/go-mysql-server.v0/sql/expression"
+	"strings"
+)
+
+// Lower is a function that returns the lowercase of the text provided.
+type Lower struct {
+	expression.UnaryExpression
+}
+
+// NewLower creates a new Lower expression.
+func NewLower(e sql.Expression) sql.Expression {
+	return &Lower{expression.UnaryExpression{Child: e}}
+}
+
+// Eval implements the Expression interface.
+func (l *Lower) Eval(
+	ctx *sql.Context,
+	row sql.Row,
+) (interface{}, error) {
+	v, err := l.Child.Eval(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+
+	if v == nil {
+		return nil, nil
+	}
+
+	if !sql.IsText(l.Child.Type()) {
+		return v, nil
+	}
+
+	switch val := v.(type) {
+	case string:
+		return strings.ToLower(val), nil
+	case []byte:
+		return []byte(strings.ToLower(string(val))), nil
+	default:
+		return val, nil
+	}
+}
+
+func (l *Lower) String() string {
+	return fmt.Sprintf("LOWER(%s)", l.Child)
+}
+
+// TransformUp implements the Expression interface.
+func (l *Lower) TransformUp(f sql.TransformExprFunc) (sql.Expression, error) {
+	child, err := l.Child.TransformUp(f)
+	if err != nil {
+		return nil, err
+	}
+	return f(NewLower(child))
+}
+
+// Type implements the Expression interface.
+func (l *Lower) Type() sql.Type {
+	return l.Child.Type()
+}
+
+// Upper is a function that returns the UPPERCASE of the text provided.
+type Upper struct {
+	expression.UnaryExpression
+}
+
+// NewUpper creates a new Lower expression.
+func NewUpper(e sql.Expression) sql.Expression {
+	return &Upper{expression.UnaryExpression{Child: e}}
+}
+
+// Eval implements the Expression interface.
+func (u *Upper) Eval(
+	ctx *sql.Context,
+	row sql.Row,
+) (interface{}, error) {
+	v, err := u.Child.Eval(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+
+	if v == nil {
+		return nil, nil
+	}
+
+	if !sql.IsText(u.Child.Type()) {
```

Comment:
According to MySQL behaviour, the argument is converted to text, then lowered. So you can just do:

```go
v, err = sql.Text.Convert(v)
if err != nil {
        return nil, err
}

return strings.ToUpper(v.(string)), nil
```